persistence/persistence-mybatis/src/main/java/org/cbioportal/persistence/mybatis/GeneticProfileMyBatisRepository.java

```diff
@@ -0,0 +1,53 @@
+package org.cbioportal.persistence.mybatis;
+
+import org.cbioportal.model.GeneticProfile;
+import org.cbioportal.model.meta.BaseMeta;
+import org.cbioportal.persistence.GeneticProfileRepository;
+import org.cbioportal.persistence.mybatis.tool.Constants;
+import org.cbioportal.persistence.mybatis.tool.OffsetCalculator;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+
+@Repository
+public class GeneticProfileMyBatisRepository implements GeneticProfileRepository {
+
+    @Autowired
+    private GeneticProfileMapper geneticProfileMapper;
+    @Autowired
+    private OffsetCalculator offsetCalculator;
+
+    @Override
+    public List<GeneticProfile> getAllGeneticProfiles(String projection, Integer pageSize, Integer pageNumber,
+                                                      String sortBy, String direction) {
+
+        return geneticProfileMapper.getAllGeneticProfiles(null, projection, pageSize,
+                offsetCalculator.calculate(pageSize, pageNumber), sortBy, direction);
+    }
+
+    @Override
+    public BaseMeta getMetaGeneticProfiles() {
+
+        return geneticProfileMapper.getMetaGeneticProfiles(null);
+    }
+
+    @Override
+    public GeneticProfile getGeneticProfile(String geneticProfileId) {
+        return geneticProfileMapper.getGeneticProfile(geneticProfileId, Constants.DETAILED_PROJECTION);
+    }
+
+    @Override
+    public List<GeneticProfile> getAllGeneticProfilesInStudy(String studyId, String projection, Integer pageSize,
+                                                             Integer pageNumber, String sortBy, String direction) {
+
+        return geneticProfileMapper.getAllGeneticProfiles(studyId, projection, pageSize,
```

Comment:
I prefer having more verbose methods in persistence-api interfaces, since they are planned to be used by 3rd parties, but we don't need that verbosity level with Mapper methods, since they are internal. So I merge the different methods in the same method in Mapper, if you provide null value to property, I ignore that property in Mapper XML file, so we have more flexible queries and less boilerplate code. But I agree that the name of the method shouldn't be `getAllGeneticProfiles` in the Mapper, it should be `getGeneticProfiles`, and if we used Java 8, we could use https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html for studyId profile, so it would be more clear that it's an optional parameter. We can change it when we switch to Java 8.